src/boxerror_replacement.rs

```diff
@@ -1,13 +1,25 @@
 //! Experimental replacement for `Box<dyn Error>` that implements `Error`.
 //!
-//! This module introduces the `DynError` type, which owns an inner 
-//! `Box<dyn Error>`. Most importantly, the `DynError` type implements the `Error`
-//! trait, unlike a plain `Box<dyn Error>`. Another important distinction between
-//! `DynError` and `Box<dyn Error>` is that `DynError` _doesn't_ implement 
-//! `From<E: Error>`. 
+//! This module introduces the `DynError` type, which owns an inner `Box<dyn Error>`. Most
+//! importantly, `DynError` _does_ implement the `Error` trait, unlike `Box<dyn Error>`. As a
+//! result, `DynError` is more conveniently compatible with the rest of the error handling
+//! ecosystem, and behaves as any other error type does. 
+//!
+//! The short answer as to why `Box<dyn Error>` doesn't implement `Error` is because there 
+//! exists a blanket implementation of the `Error` trait for `Box<T>`, more specifically: 
+//! `impl<T: Error + Sized> Error for Box<T>`. Crucially, any type `T` must be sized. However, 
+//! when `T` is a `dyn Error` trait object, it is _not_ sized. For more context on why 
+//! `Box<dyn Error>` doesn't implement the `Error` trait, see 
+//! https://stackoverflow.com/questions/65151237/why-doesnt-boxdyn-error-implement-error. 
```

Comment:
This is one step removed from the actual issue. The `?Sized` relaxed bound is missing because of the overlap it would cause between `From<T> for T` and `From<E: Error> for Box<dyn Error>` if `Box<dyn Error>` could be substituted in as `E`. So we have to prevent `Box<dyn Error>` from implementing `Error` to avoid the overlap which we do by only making it not implement `Error` when `E` is Unsized.